o1vm/src/interpreters/riscv32im/interpreter.rs

```diff
@@ -2111,14 +2108,52 @@ pub fn interpret_sbtype<Env: InterpreterEnv>(env: &mut Env, instr: SBInstruction
     };
     env.range_check8(&rs2, 5);
 
-    // FIXME: trickier
-    let imm2 = {
-        let pos = env.alloc_scratch();
-        unsafe { env.bitmask(&instruction, 32, 25, pos) }
-    };
-    env.range_check16(&imm2, 12);
+    let imm0_12 = {
+        let imm11 = {
+            let pos = env.alloc_scratch();
+            unsafe { env.bitmask(&instruction, 8, 7, pos) }
+        };
 
-    // FIXME: check correctness of decomposition
+        env.range_check8(&imm11, 1);
```

Comment:
It seems strange that we check that a single "bit" is in range `[0,255]`, or am i misunderstanding the point?